### ISSUE TITLE
livecd-create empties(!) .ks config if it's local

### ISSUE THREAD

**Issue by asashnov** [livecd-tools/livecd-tools#155]
Git master version: 0703f6c7

Configuration .ks file is local, in the same directory. Running empties the config file:

```
 # livecd-create -c keepod-srw-os.ks
Complains:
Kickstart (keepod-srw-os.ks) must have at least one repository.

# cat keepod-srw-os.ks
#    <=== now it's empty!
```

### Additional details

In kickstart.py:
```python
        ksfile = urlgrabber.urlgrab(path)      # <===  this line empties the file.
```

This is because /usr/lib/python3.7/site-packages/urlgrabber/grabber.py opens the file with 'wb' mode.

Example:

open.py:

#!/usr/bin/python3.7
import urlgrabber
path = "file"
f = urlgrabber.urlgrab(path)

[root@a4595b22b103 keepodsrwos]# echo hello > file
[root@a4595b22b103 keepodsrwos]# cat file
hello
[root@a4595b22b103 keepodsrwos]# python3.7 open.py
[root@a4595b22b103 keepodsrwos]# cat file

Additional details:

I experimented inside official Docker Fedora 31 container and even tried to update UrlGrabber:

```
# dnf install python3-pip
# pip-3.7 install -U urlgrabber
```

But the result is the same.

open.py:

```python
#!/usr/bin/python3.7
import urlgrabber
path = "file"
f = urlgrabber.urlgrab(path)
```

```
[root@a4595b22b103 keepodsrwos]# echo hello > file
[root@a4595b22b103 keepodsrwos]# cat file
hello
[root@a4595b22b103 keepodsrwos]# python3.7 open.py
[root@a4595b22b103 keepodsrwos]# cat file
[root@a4595b22b103 keepodsrwos]#    <====  emptied!!!
```

**Comment by Conan-Kudo** [comment]
@asashnov Could you please file a bug with [urlgrabber](https://github.com/rpm-software-management/urlgrabber)?

**Comment by asashnov** [comment]
Upstream issue created:
https://github.com/rpm-software-management/urlgrabber/issues/26

**Comment by sobjerke** [comment]
I think the problem is that urlgrabber.urlgrab (without filename= argument) overwrites the file if it is located in the current directory. I used the following patch to get around the problem:

--- a/imgcreate/kickstart.py
+++ b/imgcreate/kickstart.py
@@ -51,8 +51,10 @@ def read_kickstart(path):
     version = ksversion.makeVersion()
     ks = ksparser.KickstartParser(version)
     try:
-        ksfile = urlgrabber.urlgrab(path)
-        ks.readKickstart(ksfile)
+        tmpks = '.kstmp.{}'.format(os.getpid())
+        ksfile = urlgrabber.urlgrab(path, filename=tmpks)
+        ks.readKickstart(tmpks)
+        os.unlink (tmpks)

When using a local file, I cannot see why it should be downloaded at all (other protocols fails anyway since os.path.exists is used on the config argument) .

**Comment by Conan-Kudo** [comment]
@sobjerke Can you submit this as a PR? I'd be happy to take this to fix it. 👍🏾

**Comment by sobjerke** [comment]
@Conan-Kudo PR added

**Comment by Conan-Kudo** [comment]
This is now fixed with f711b0de840a012b74ddc792557a49cecf6e4bc5 from @sobjerke!
